apps/wallet/src/features/Swap/InputSwap/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { type FC, useState } from 'react';
+import { ActivityIndicator, StyleSheet } from 'react-native';
+import type { SlideComponentProps } from '@walless/gui';
+import { Button, View } from '@walless/gui';
+import ModalHeader from 'components/ModalHeader';
+import { showError } from 'modals/Error';
+import { usePublicKeys, useSnapshot } from 'utils/hooks';
+
+import { swapActions, swapContext } from '../context';
+
+import FromToken from './FromToken';
+import SwitchSeparator from './SwitchSeparator';
+import ToToken from './ToToken';
+
+type Props = SlideComponentProps;
+
+const InputSwap: FC<Props> = ({ navigator }) => {
+	const { network } = useSnapshot(swapContext).swap;
+	const publicKeys = usePublicKeys(network);
+	const [loading, setLoading] = useState(false);
+
+	const handlePressSwap = async () => {
+		setLoading(true);
+		try {
+			await swapActions.prepareSwapTransaction(publicKeys[0]._id);
+			navigator.slideNext();
+		} catch (error) {
+			showError({ errorText: (error as Error).message });
+		}
+		setLoading(false);
+	};
+
+	const handleClose = () => {
+		swapActions.closeSwap();
+		setTimeout(() => {
+			swapActions.resetContext();
+		}, 200);
```

Comment:
why do we have to wait for 200ms to reset the context? it seems like a magic number

Reply:
yep, It's a magic number. But the problem comes from the animation behavior of the modal component, it should expose life cycle methods to control when to remove context correctly. I keep the code, but will mark it to resolve in an issue